airflow/providers/trino/hooks/trino.py

```diff
@@ -0,0 +1,191 @@
+from airflow.hooks.dbapi import DbApiHook
+from airflow.models import Connection
+
+
+class TrinoException(Exception):
+    """Trino exception"""
+
+
+def _boolify(value):
+    if isinstance(value, bool):
+        return value
+    if isinstance(value, str):
+        if value.lower() == 'false':
+            return False
+        elif value.lower() == 'true':
+            return True
+    return value
+
+
+class TrinoHook(DbApiHook):
+    """
+    Interact with Trino through trino package.
+
+    >>> ph = TrinoHook()
+    >>> sql = "SELECT count(1) AS num FROM airflow.static_babynames"
+    >>> ph.get_records(sql)
+    [[340698]]
+    """
+
+    conn_name_attr = 'trino_conn_id'
+    default_conn_name = 'trino_default'
+    conn_type = 'trino'
+    hook_name = 'Trino'
+
+    def get_conn(self) -> Connection:
+        """Returns a connection object"""
+        db = self.get_connection(
+            self.trino_conn_id  # type: ignore[attr-defined]  # pylint: disable=no-member
+        )
+        extra = db.extra_dejson
+        auth = None
+        if db.password and extra.get('auth') == 'kerberos':
+            raise AirflowException("Kerberos authorization doesn't support password.")
+        elif db.password:
+            auth = trino.auth.BasicAuthentication(db.login, db.password)
+        elif extra.get('auth') == 'kerberos':
+            auth = trino.auth.KerberosAuthentication(
+                config=extra.get('kerberos__config', os.environ.get('KRB5_CONFIG')),
+                service_name=extra.get('kerberos__service_name'),
+                mutual_authentication=_boolify(extra.get('kerberos__mutual_authentication', False)),
+                force_preemptive=_boolify(extra.get('kerberos__force_preemptive', False)),
+                hostname_override=extra.get('kerberos__hostname_override'),
+                sanitize_mutual_error_response=_boolify(
+                    extra.get('kerberos__sanitize_mutual_error_response', True)
+                ),
+                principal=extra.get('kerberos__principal', conf.get('kerberos', 'principal')),
+                delegate=_boolify(extra.get('kerberos__delegate', False)),
+                ca_bundle=extra.get('kerberos__ca_bundle'),
+            )
+
+        trino_conn = trino.dbapi.connect(
+            host=db.host,
+            port=db.port,
+            user=db.login,
+            source=db.extra_dejson.get('source', 'airflow'),
+            http_scheme=db.extra_dejson.get('protocol', 'http'),
+            catalog=db.extra_dejson.get('catalog', 'hive'),
+            schema=db.schema,
+            auth=auth,
+            isolation_level=self.get_isolation_level(),  # type: ignore[func-returns-value]
+        )
+        if extra.get('verify') is not None:
+            # Unfortunately verify parameter is available via public API.
+            # The PR is merged in the trino library, but has not been released.
+            # See: https://github.com/trinosql/trino-python-client/pull/31
```

Comment:
```suggestion
            # Unfortunately verify parameter is available via public API.
            # The PR is merged in the trino library, but has not been released.
            # See: https://github.com/trinodb/trino-python-client/pull/31
```
BTW this PR is merged and released